Adjust the <AccountCard> props to make their usages clearer and tweak some wordings for GMC cards

eason9487 opened the pull request:
### Changes proposed in this Pull Request:

It's a follow-up PR from https://github.com/woocommerce/google-listings-and-ads/pull/1102#discussion_r753686922, and based on PR #1119.

- Adjust the `<AccountCard>` props to make their usages clearer and apply them on custom appearance cards.
    - Contact information preview card
    - Default WordPress account card
    - Loading state of Google account card
- Add default description for Google Merchant Center card appearance and apply it on relevant cards.
    - Create account card
    - Connect account card
    - Disabled card
- Tweak the button wording of the disabled Google Merchant Center card.

### Additional note


I added a demo for the sake of easy viewing of the various ways to use the `<AccountCard>`

- [x] :warning: Revert https://github.com/woocommerce/google-listings-and-ads/commit/ec10fb48841e549ebf012eeef43b482432fd2b4c before merging this PR.

### Screenshots:

#### Contact information preview card

![2021-11-29 11 39 21](https://user-images.githubusercontent.com/17420811/143810148-61bffe5a-20ad-4813-a7c2-120afe05f9f0.png)

https://user-images.githubusercontent.com/17420811/143810086-5a8d5f4e-beab-46c3-b023-905aa96a4712.mp4

#### Default WordPress account card

![image](https://user-images.githubusercontent.com/17420811/143810583-6f45931e-1ba1-4773-bedd-bdf1e624f59f.png)

#### Loading state of Google account card

![image](https://user-images.githubusercontent.com/17420811/143810620-375fb35c-5bd8-4786-9e13-c3d1a524e4d3.png)

#### Default description and adjusted wordings of Google Merchant Center cards

![2021-11-29 11 53 53](https://user-images.githubusercontent.com/17420811/143810176-77bf1a1e-69a0-40f7-98f6-edc38e9f0824.png)

![2021-11-29 11 55 23](https://user-images.githubusercontent.com/17420811/143810184-9181bdf5-1485-4f9e-9961-cc28a25bf2cb.png)

![2021-11-29 11 54 47](https://user-images.githubusercontent.com/17420811/143810183-6a309f1a-d2de-48be-8e13-9f047875a540.png)

### Detailed test instructions:

💡  Currently, we only have the Google account reconnection flow that might show [the loading Google account card](https://github.com/woocommerce/google-listings-and-ads/blob/8af03363a9b501c47c4e3b0c52896d131d0c4900/js/src/components/google-account-card/google-account-card.js#L14). But in addition to disconnecting your Google account and entering the reconnection process, and allowing only partial authorization during the process, then you would have the chance to enter this logic.

Due to the tedious process of adjusting the connection status of Google account and the API response time gap, it will take much effort and still has a very low chance of reaching that condition. Instead, in step 3 of test instructions, you could view the last demo card, which is created in the same code, on the GLA Settings page to check if it shows correctly.

1. Go to the onboarding setup to see if the WordPress account card is displayed correctly as before.
2. Go to the GLA Settings page to see if the preview of contact information is displayed correctly as before.
    - To check the warning state, you could return a mocking data before line 519 at [actions.js](https://github.com/woocommerce/google-listings-and-ads/blob/8af03363a9b501c47c4e3b0c52896d131d0c4900/js/src/data/actions.js#L518-L523) with:
    ```js
    return {
    	type: TYPES.RECEIVE_MC_CONTACT_INFORMATION,
    	data: {
    		id: 123,
    		phone_number: '',
    		is_mc_address_different: true,
    		mc_address: {},
    		wc_address: {},
    	},
    };
    ```
3. Check all demo cards on the GLA Settings page.
    ![image](https://user-images.githubusercontent.com/17420811/143816975-2d2b4bea-ce4c-403f-a850-34b5b55bd800.png)

### Changelog entry

eason9487 committed:
Adjusting the <AccountCard> props to make their usages clearer

eason9487 committed:
Use adjusted props of <AccountCard> to custom appearance

eason9487 committed:
Add default description for GMC card appearance and apply it on relevant cards

eason9487 committed:
Tweak the button wording of disabled GMC card

eason9487 committed:
Add a temporary demo for <AccountCard> component

eason9487 commented on js/src/setup-mc/setup-stepper/setup-accounts/google-mc-account/disabled-card.js:
```diff
@@ -14,13 +14,9 @@ const DisabledCard = () => {
 			indicator={
 				<Button isSecondary disabled>
-					{ __( 'Connect', 'google-listings-and-ads' ) }
+					{ __( 'Create Account', 'google-listings-and-ads' ) }
```
When checking the implementations against visual design, I found this button text displays "Create Account". So I adjusted it by the way.

ecgan commented on js/src/components/account-card/index.js:
```diff
@@ -75,33 +81,33 @@ const alignStyleName = {
-	appearance,
 	disabled = false,
-	description,
+	appearance = 'empty',
```
😄 

```suggestion
	appearance = APPEARANCE.EMPTY,
```

eason9487 replied:
Nice catch! Thanks!

ecgan commented on js/src/components/account-card/index.js:
```diff
@@ -75,33 +81,33 @@ const alignStyleName = {
 	disabled = false,
-	description,
+	appearance = 'empty',
+	icon = appearanceDict[ appearance ].icon,
+	title = appearanceDict[ appearance ].title,
+	description = appearanceDict[ appearance ].description,
```
❔ 💅 This relates back to my comment in https://github.com/woocommerce/google-listings-and-ads/pull/1102#discussion_r753686922 about "on the consumer perspective, it starts to feel a bit funny because the consumer can provide description via `description` props, or `appearance` props like `appearance={{ description: '' }}`."

I just thought about this idea: what about if we make those `APPEARANCE.GOOGLE` an individual object in their own file, like this:

```js
// file: appearances/google.js
// let consumer get access to the appearance object, 
// instead of appearance string.
export default {
	icon: googleLogo,
	title: __( 'Google Ads', 'google-listings-and-ads' ),
	description: __(
		'Required to create paid campaigns with your product listings',
		'google-listings-and-ads'
	),
};
```

Then, on the consumer side, we can do this:

```js
// import the object from above.
import appearanceGoogle from './appearances/google.js'

// use it via `appearance` prop.
// with custom description.
// don't need icon, title, description props.
const card1 = <AccountCard
	appearance={ { ...appearanceGoogle, description: 'Hello!' } }
/>

// Or use it via AccountCard props: icon, title, description.
// with custom description.
// don't need appearance props.
// Personally, I'm leaning towards this.
const card2 = <AccountCard
	{ ... appearanceGoogle }
	description='Hello!'
/>
```

With the above approach, we only need `appearance` props, or the "icon, title, description" props. Also, we can remove `appearanceDict` too.

eason9487 replied:
> This relates back to my comment in https://github.com/woocommerce/google-listings-and-ads/pull/1102#discussion_r753686922 about "on the consumer perspective, it starts to feel a bit funny because the consumer can provide description via `description` props, or `appearance` props like `appearance={{ description: '' }}`."

I thought your former concern was there were two methods to set card `description` and it brought some confusion. By this PR, it changed to have only one way to specify the `description` from the consumer side, and the `appearance` was changed to specify the default and highly reused card appearance only. The concern itself should have already been addressed?

> I just thought about this idea: what about if we make those `APPEARANCE.GOOGLE` an individual object in their own file
[...]
With the above approach, we only need `appearance` props, or the "icon, title, description" props. Also, we can remove `appearanceDict` too.

I don't see the `appearanceDict` causing any problems, and I'm not sure about the benefits of making them standalone files for the present. In addition, using `appearance` with props object spreading in every `<AccountCard>` makes them more tedious to read and develop. In my opinion, under the premise of high reuse base props, internalizing them is relatively simple and easy to use. And it already supports the `card2` method as well.

ecgan replied:
> the `appearance` was changed to specify the default and highly reused card appearance only.

Sorry, my bad, I just realized the `appearance` prop has been changed from accepting an object to now accepting a string. 🤦 😅

ecgan reviewed (approved):
Tested, working as expected. Left two non-blocking comments below, one with code suggestion. LGTM 👍

eason9487 committed:
Use `APPEARANCE.EMPTY` instead of a duplicate string declaration for default prop value

Address https://github.com/woocommerce/google-listings-and-ads/pull/1120#discussion_r758103517

Co-authored-by: Gan Eng Chin <ecgan@users.noreply.github.com>

eason9487 commented:
> :warning: Revert ec10fb4 before merging this PR.

Totally forgot to revert it. 🤦‍♂️  Reverted it directly to the shared branch `feature/1065-ux-account-connection` by https://github.com/woocommerce/google-listings-and-ads/commit/c3172f2ae8995ef54b264138162656a255f9523a. 🙏
